rfcs/convergence/react-focus-management.md

```diff
@@ -0,0 +1,169 @@
+
+- Force rerender items that aren't visible
+- Cause unnecessary network requests due to force rerendering
+- Extra narration for screen reader users with unnecessary items not rendered
+
+Nested focusables are a source of issues on the current `FocusZone` component, where nested focusables behave inconsistently or incorrectly. Here are some recent issues related to this
+
+- [#8551](https://github.com/microsoft/fluentui/issues/8551)
+- [#12177](https://github.com/microsoft/fluentui/issues/12177)
+- [#16037](https://github.com/microsoft/fluentui/issues/16037)
+- [#13210](https://github.com/microsoft/fluentui/issues/13210)
+
+## Detailed Design or Proposal
+
+### Ability helpers
+
+<!-- This is the bulk of the RFC. Explain the proposal or design in enough detail for the inteded audience to understand. -->
+
+The `@fluentu/react-focus-management` package will act as a facade wrapper around [ability-helpers](https://github.com/microsoft/ability-helpers) and exposes a set of hooks and utilities for focus management. The [ability-helpers](https://github.com/microsoft/ability-helpers) library is designed to be a lower level utility that manages the state of focusable elements than any rendering framework.
+
+[ability-helpers](https://github.com/microsoft/ability-helpers) manages focus on a lower level than the rendering framework and acts on the **visible** DOM. Since operations are run on visible DOM there is a much clearer path for handling dynamically loaded content that would otherwise require effects outside the rendering lifecycle in React components which breaks component isolation principles.
+
+Patterns in [ability-helpers](https://github.com/microsoft/ability-helpers) are **declarative**, and therefore decoupled more easily with framework and application code. Groups of focusables are managed by declaring `data-*` attributes which the lower level utilties use to define focusable behaviour.
+
+Nesting focusables is a problem that ability-helpers explicitly tries to solve, bringing focus management to a lower level than the rendering framework tries to accomplish this consistently independent of rendering cycles.
+
+### Initial API
+
+#### FocusManagementProvider
+
+Context provider that hosts the [ability-helpers](https://github.com/microsoft/ability-helpers) instance and its API objects.
+
+The below table defines what is included in the context value
+
+> Note that this table will increase as more ability-helpers features are included in Fluent
+
+| Name      | Value        | Description                           |
+| --------- | ------------ | ------------------------------------- |
+| focusable | FocusableAPI | API for traversing focusable elements |
+
+The below table defines the props used by the provider component. All context values can be used as props in the provider component but are optional.
```

Comment:
I don't think that it's a good idea to expose `internal__ahInstance` as a prop 🐱 Can you please add a code snippet to better understand difference between tables?

```tsx
interface AhContextValue {
  // something is there
}

const AhContext = React.createContext<>()

//

function App() {
  return <AhProvider {/* props that can be defined are there */} />
}
```

Reply:
Agreed, exposing anything `internal` seems iffy to me.